product_multi_company/tests/test_product_multi_company.py

```diff
@@ -77,3 +79,9 @@ def test_uninstall(self):
         domain = (" ['|',('company_id','=',user.company_id.id),"
                   "('company_id','=',False)]")
         self.assertEqual(rule.domain_force, domain)
+
+    def test_init_hook(self):
+        deactivated_product = self.env.ref('product.product_product_4d')
```

Comment:
While the product is inactive, its template is not inactive so it does not test what is implied. How about Pedro's suggestion, to inject some data at this point and then call the hook method again?

Reply:
The SQL in that suggestion is quite intense due to M2M relation. I'll look into it though yeah

Reply:
Well, I think it's not very difficult:
```python
product = env['product.product'].create({
    'name': 'Test',
    'active': False,
})
env.cr.execute(
    "DELETE FROM <table_rel> "
    "WHERE product_id = %s",
    (product.id, ),
)
<call_hook>()
# Check that product.company_ids is filled.
```